src/Domain/ShoppingList.php

```diff
@@ -80,6 +87,27 @@ public function getItems(): ShoppingItemStack
         return $this->items;
     }
 
+    /**
+     * Get whether an item has been archived.
+     *
+     * @return bool
+     */
+    public function isArchived(): bool
+    {
+        return $this->isArchived;
+    }
+
+    /**
+     * Set whether an item is archived.
+     *
+     * @param bool $isArchived
+     * @return void
+     */
+    public function setArchived(bool $isArchived): void
```

Comment:
Required: good job on adding the setter, this is correct! Just one change required. The `setName()` method does `return $this`. We'd expect setters on the class to be consistent, so you should also do the same for this method.